store/SPEC_STANDARD.md

```diff
@@ -0,0 +1,110 @@
+## What is a store specification document?
+
+A store specification document is a design document describing a particular store feature such as a store interface or implementation that it is expected to be used (or already in use) by the Cosmos SDK. This also includes other store-related features such as caches and wrappers.
+
+## Sections
+
+An store specification document consists of, synopsis, overview and basic concepts, technical specification, history log, and copyright notice. All top-level sections are required. References should be included inline as links, or tabulated at the bottom of the section if necessary.  Included sub-sections should be listed in the order specified below. 
+
+### Synopsis
+
+The document should include a brief (~200 word) synopsis providing a high-level
+description of and rationale for the specification.
+
+### Overview and basic concepts
+
+This section should include a motivation sub-section and a definitions sub-section if required:
+
+- *Motivation* - A rationale for the existence of the proposed interface, implementation, or the proposed changes to an existing feature.
+- *Definitions* - A list of new terms or concepts utilised in the document or required to understand it.
+
+### System model and properties
+
+This section should include an assumptions sub-section if any, and the mandatory properties sub-section. Note that both sub-sections are tightly coupled: how to enforce a property will depend directly on the assumptions made.
```

Comment:
I think the important thing to capture here is the "rest of the world", that is, 
1. Which modules are used by the module under specification, and what do we expect from them
2. If possible, which module use the module under specification and how.

For instance, let's assume module A uses module B. There might be the implicit assumption that module B panics in certain situations, and this is fine, as we want A to panic in this situations as well. This is an assumption we need to make explicit in the specification.

With this information, we obtain:
a. We understand what A is doing as a result of the behavior of B
b. If we ever go to redesign B, we need to understand that other modules (e.g. A) rely on maintaining this specific behavior (e.g., we still need to panic in this situation).

I guess Point 1. would go into the assumption paragraph and Point 2 in the property paragraph.